deepspeed/runtime/zero/utils.py

```diff
@@ -67,6 +67,23 @@ def is_zero_supported_optimizer(optimizer):
     return type(optimizer) in ZERO_SUPPORTED_OPTIMIZERS
 
 
+@instrument_w_nvtx
+def assert_lst_len_same_as_other_ranks(lst: List[int]) -> None:
+    rank0_len_tensor = torch.tensor(
+        len(lst) if dist.get_rank() == 0 else -1,
+        dtype=int,
+        device=torch.device(get_accelerator().device_name(os.environ["LOCAL_RANK"])),
+        requires_grad=False,
+    )
+    local_len_tensor = len(lst)
+    dist.broadcast(rank0_len_tensor, src=0, async_op=False)
+    rank0_len_tensor_cpu = rank0_len_tensor.cpu().item()
+    if rank0_len_tensor_cpu != local_len_tensor:
+        raise RuntimeError(f"disagreement between rank0 and rank{dist.get_rank()}: "
```

Comment:
```suggestion
        raise RuntimeError(f"Detected a disagreement between rank0 and rank{dist.get_rank()}: "
```

Reply:
and the same in the other instance of this message - line 97 in the original, 115 in this PR